addon/index.js

```diff
@@ -12,7 +12,64 @@
  * ```
  */
 export function proto(target, key, desc) {
-  desc.value = desc.initializer();
+  desc.value = desc.initializer.apply(target);
   desc.initializer = undefined; // unset the initializer
   return desc;
 }
+
+/**
+ * Decorator which merges the value of the class field with the value
+ * of the field on the superclass.
+ *
+ * ```js
+ * class Foo {
+ *   baz = { prop1: 123 };
+ * }
+ *
+ * class Bar {
+ *   @concatenated baz = { prop2: 456 };
+ * }
+ *
+ * let bar = new Bar();
+ * console.log(bar.baz); // { prop1: 123, prop2: 456 }
+ * ```
+ */
+export function merged(target, key, desc) {
+  const valuesToMerge = desc.initializer();
```

Comment:
Eagerly calling this initializer means that it cannot use `this`, we could fix that by making it lazy.

```js
const specifiedInitializer = desc.initializer;
desc.initializer = function() {
  const merged = {};
  const originalValues = this[key];
  const valuesToMerge = specifiedInitializer.call(this);
  
  return Object.assign(merged, originalValues, valuesToMerge);
};
```

Reply:
mm, my logic was caching the value once would be more performant, but that's a great point. I'll update it